products/datastore/api.yaml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Datastore
+display_name: Cloud Datastore
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://datastore.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/datastore
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Datastore APU
+    url: https://console.cloud.google.com/apis/library/datastore.googleapis.com
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+
+objects:
+  - !ruby/object:Api::Resource
```

Comment:
You'll also want to add a `references: !ruby/object:Api::Resource::ReferenceLinks` section, not sure why we don't require that field